base/loading.jl

```diff
@@ -471,26 +804,23 @@ include_string(m::Module, txt::String, fname::String) =
 include_string(m::Module, txt::AbstractString, fname::AbstractString="string") =
     include_string(m, String(txt), String(fname))
 
-function source_path(default::Union{AbstractString,Nothing}="")
+function tls_recurse(key::Symbol, default=nothing)
```

Comment:
I was at some point trying passing ENVINFO via task local storage, which is why I refactored this, but I didn't end up using it. I'm leaving the refactoring in since it's good to separate the TLS recursion from source path lookup anyway.

Reply:
This method is deprecated (v0.7) anyways, I'm not sure it matters